README.md

```diff
@@ -0,0 +1,115 @@
+# DeltaKernel Change Impact Analysis Tool
+
+## Table of Content
+
+- [Introduction](#introduction)
+- [Innovation](#innovation)
+- [How to Use](#how-to-use)
+- [Files Generated](#files-generated)
+- [Structure of the Tool](#structure-of-the-tool)
+  - [I. Compilation File List Generation](#i-compilation-file-list-generation)
+  - [II. Git Diff Report Generation](#ii-git-diff-report-generation)
+  - [III. Commit Metadata Retrieval](#iii-commit-metadata-retrieval)
+  - [IV. Web Script Generation](#iv-web-script-generation)
+
+## Introduction
+
+DeltaKernel Change Impact Analysis Tool generates a comprehensive visual report detailing changes in both header files and source code between two Linux versions (tags in the Linux kernel repository: old_tag and new_tag). This tool helps developers view updates from the old version.
+
+The diff report includes a subset of files from the Linux repository that are included in building the kernel, contributing to a focused and detailed report on the compile-time source code in Linux.
+
+## Innovation
+
+The idea of generating a web display for Linux kernel version change impact analysis is inspired by [Cregit](https://github.com/cregit/cregit). This tool innovates on Cregit by:
+
+- Considering the extensive code space the Linux kernel deals with, it provides a compile-time analysis instead of a static analysis of the commit history of the Linux kernel, presenting changes only in files used during compilation. 
```

Comment:
The tool does not allow user to define kernel configuration. "compile-time" means it do a version analysis (like git diff) only for files that are used to compile the kernel (instead of the whole Linux source code space). I have modified that documentation for more clarification.